web/src/compose_validate.ts

```diff
@@ -813,55 +853,76 @@ export function check_overflow_text($container: JQuery): number {
     return text.length;
 }
 
-export function validate_message_length($container: JQuery): boolean {
+export function validate_message_length($container: JQuery, show_banner = true): boolean {
     const $textarea = $container.find<HTMLTextAreaElement>(".message-textarea");
     // Match the behavior of compose_state.message_content of trimming trailing whitespace
     const text = $textarea.val()!.trimEnd();
     const message_too_long_for_compose = text.length > realm.max_message_length;
     set_message_too_long_for_compose(message_too_long_for_compose);
     if (message_too_long_for_compose) {
-        $textarea.addClass("flash");
-        setTimeout(() => $textarea.removeClass("flash"), 1500);
-        const $banner_container = $("#compose_banners");
-        compose_banner.show_error_message(
-            get_message_too_long_for_compose_error(),
-            compose_banner.CLASSNAMES.exceeded_message_length_limit,
-            $banner_container,
-            $("#message-content-container"),
-        );
+        report_validation_error(() => {
+            $textarea.addClass("flash");
+            setTimeout(() => $textarea.removeClass("flash"), 1500);
+
+            const $banner_container = $("#compose_banners");
+            compose_banner.show_error_message(
+                get_message_too_long_for_compose_error(),
+                compose_banner.CLASSNAMES.exceeded_message_length_limit,
+                $banner_container,
+                $("#message-content-container"),
+            );
+        }, show_banner);
         return false;
     }
     return true;
 }
 
-export function validate(scheduling_message: boolean): boolean {
+function report_validation_error(show_error_func: () => void, show_banner: boolean): void {
+    if (show_banner) {
+        show_error_func();
+    }
+}
+export function validate(
+    scheduling_message: boolean,
+    show_banner = true,
+    update_send_button_disability = false,
+): boolean {
     const message_content = compose_state.message_content();
     // The validation checks in this function are in a specific priority order. Don't
     // change their order unless you want to change which priority they're shown in.
+    function handleReturn(value: boolean): boolean {
+        if (update_send_button_disability) {
+            update_send_button_status();
+        }
```

Comment:
would it be better to just move this to compose_setup?

Reply:
Since `update_send_button_disability` is only `true` inside `compose_setup`, we can just move this call there and avoid using  `handleReturn`.